acasclient/lsthing.py

```diff
@@ -584,16 +584,45 @@ class BlobValue(object):
     These files must be small (< 1 GB) and will be stored in a `bytea` database column.
     """
 
-    def __init__(self, value=None, comments=None):
+    def __init__(self, value=None, comments=None, id=None, ls_value=None):
         """Create a BlobValue
 
         :param value: Bytes of file content, defaults to None
         :type value: bytes, optional
         :param comments: Filename as a string, defaults to None
         :type comments: str, optional
+        :param id: id as an int, defaults to None
+        :type id: int, optional
+        :param ls_value: ls_value as a <acasclient.lsthing.LsThingValue>, defaults to None
+        :type ls_value: <acasclient.lsthing.LsThingValue>, optional
         """
+        if ls_value is not None:
+            value = ls_value.blob_value
+            comments = ls_value.comments
+            id = ls_value.id
+        else:
+            if value is not None:
+                if isinstance(value, Path) or isinstance(value, str):
+                    if isinstance(value, str):
+                        value = Path(value)
+                    if comments is None:
+                        comments = value.name
+                    f = value.open('rb')
+                    bytes_array = f.read()
+                    value = [x for x in bytes_array]
+                    f.close()
+
         self.value = value
         self.comments = comments
+        self.id = id
+
+    def get_data(self, client):
```

Comment:
nit: `get_data` is a little vague and doesn't clearly say that this is fetching from the ACAS server versus just fetching a value from local memory.
Perhaps `get_blob_data_from_acas` if we want to be verbose, or `download_data` ?

Reply:
I like `download_data`.  I'll go with that.